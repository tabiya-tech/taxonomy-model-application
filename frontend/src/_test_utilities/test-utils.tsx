// Based on https://testing-library.com/docs/react-testing-library/setup/

import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "@mui/material";
import applicationTheme, { ThemeMode } from "src/theme/applicationTheme/applicationTheme";
import SnackbarProvider from "src/theme/SnackbarProvider/SnackbarProvider";
import { IsOnlineProvider } from "src/app/providers";

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <IsOnlineProvider>
      <ThemeProvider theme={applicationTheme(ThemeMode.LIGHT)}>
        <SnackbarProvider>{children}</SnackbarProvider>
      </ThemeProvider>
    </IsOnlineProvider>
  );
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "wrapper">) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
