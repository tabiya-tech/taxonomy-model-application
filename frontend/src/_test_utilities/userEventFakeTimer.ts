import userEvent from "@testing-library/user-event";
import { act } from "@testing-library/react";
import { UserEvent } from "@testing-library/user-event/setup/setup";

async function setup(callback: (user: UserEvent) => Promise<void>) {
  // We need to use fake times to ensure the debounce timer is progressed,
  // Since the debounce timer is called after the user stops typing, we need to set the fake timers
  // before the user begins typing, and for this reason we need to set up the userEvent to advance the timers using the jest timers,
  // so that the await user.type() can be resolved.
  // User enters a model name
  jest.useFakeTimers();
  const userEventFakeTimer = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
  await callback(userEventFakeTimer);
  act(() => jest.runOnlyPendingTimers());
  jest.useRealTimers();
}

export async function typeDebouncedInput(element: Element, text: string) {
  await setup(async (user) => {
    await user.type(element, text);
  });
}

export async function clickDebouncedButton(element: Element) {
  await setup(async (user) => {
    await user.click(element);
  });
}
