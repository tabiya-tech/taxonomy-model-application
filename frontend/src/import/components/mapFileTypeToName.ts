import ImportAPISpecs from "api-specifications/import";

export const mapFileTypeToName = (fileType: ImportAPISpecs.Constants.ImportFileTypes): string => {
  switch (fileType) {
    case ImportAPISpecs.Constants.ImportFileTypes.ISCO_GROUPS:
      return "ISCO Groups";
    case ImportAPISpecs.Constants.ImportFileTypes.OCCUPATIONS:
      return "Occupations";
    case ImportAPISpecs.Constants.ImportFileTypes.ESCO_SKILLS:
      return "Skills";
    case ImportAPISpecs.Constants.ImportFileTypes.ESCO_SKILL_GROUPS:
      return "Skill Groups";
    case ImportAPISpecs.Constants.ImportFileTypes.ESCO_SKILL_SKILL_RELATIONS:
      return "Skill-To-Skill Relations";
    case ImportAPISpecs.Constants.ImportFileTypes.OCCUPATION_HIERARCHY:
      return "Occupation Hierarchy";
    case ImportAPISpecs.Constants.ImportFileTypes.ESCO_SKILL_HIERARCHY:
      return "Skill Hierarchy";
    case ImportAPISpecs.Constants.ImportFileTypes.OCCUPATION_SKILL_RELATIONS:
      return "Occupation-To-Skill Relations";
    // case ImportAPISpecs.Constants.ImportFileTypes.OCCUPATION_LOGS:
    //   return "Occupation Logs";
    // case ImportAPISpecs.Constants.ImportFileTypes.OCCUPATION_LOG_CHANGES:
    //   return "Occupation Log Changes";
    default:
      return "Unknown File Type";
  }
};
