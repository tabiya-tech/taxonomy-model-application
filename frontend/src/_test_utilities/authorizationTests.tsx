import { render, screen } from "src/_test_utilities/test-utils";
import AuthAPISpecs from "api-specifications/auth";
import { mockLoggedInUser } from "./mockLoggedInUser";

export const ALL_USERS = [
  AuthAPISpecs.Enums.TabiyaRoles.ANONYMOUS,
  AuthAPISpecs.Enums.TabiyaRoles.REGISTERED_USER,
  AuthAPISpecs.Enums.TabiyaRoles.MODEL_MANAGER,
];

type AuthorizationTestsParams = {
  name: string;
  Component: JSX.Element;
  testIds?: string[];
  roles?: AuthAPISpecs.Enums.TabiyaRoles[];
  tests?: Partial<
    Record<
      AuthAPISpecs.Enums.TabiyaRoles,
      {
        visible?: string[];
        hidden?: string[];
        visibleDisabled?: string[];
      }
    >
  >;
};

export const authorizationTests = {
  defaultName: "Test the rendering of the component based on Roles",
  callback: ({ name, Component, testIds = [], roles = ALL_USERS, tests }: AuthorizationTestsParams) => {
    return () => {
      test.each(roles?.map((role) => [role]))(`${name} renders correctly for user with role %s`, (role) => {
        const commonElements = testIds || [];
        const { visible = [], visibleDisabled = [], hidden = [] } = tests?.[role] || {};

        mockLoggedInUser({ user: { username: "foo", roles: [role] } });

        render(Component);

        commonElements.forEach((testId) => expect(screen.getByTestId(testId)).toBeInTheDocument());

        visible?.forEach((testId) => {
          expect(screen.getByTestId(testId)).toBeVisible();
        });

        hidden?.forEach((testId) => {
          expect(screen.queryByTestId(testId)).not.toBeInTheDocument();
        });

        visibleDisabled?.forEach((testId) => {
          const element = screen.getByTestId(testId);
          expect(element).toBeVisible();
          if (element instanceof HTMLInputElement) {
            expect(element).toBeDisabled();
          }
        });
      });
    };
  },
};
