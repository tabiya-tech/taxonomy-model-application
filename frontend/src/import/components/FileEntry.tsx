import * as React from "react";
import {ChangeEvent, useState} from "react";
import AddIcon from "@mui/icons-material/AddCircleOutline";
import RemoveIcon from "@mui/icons-material/RemoveCircleOutline";
import * as Import from "api-specifications/import";
import {Fab, FabProps} from "@mui/material";
import {generateUniqueId} from "src/utils/generateUniqueId";
import {mapFileTypeToName} from "./mapFileTypeToName";
import debounce from "lodash.debounce";
import {DEBOUNCE_INTERVAL} from "./debouncing";

export interface FileEntryProps {
  fileType: Import.Types.ImportFileTypes,
  notifySelectedFileChange?: (fileType: Import.Types.ImportFileTypes, newFile: File | null) => void
}

const baseTestID = "d2bc4d5d-7760-450d-bac6-a8857affeb89"

export const DATA_TEST_ID = {
  FILE_ENTRY: `file-entry-${baseTestID}`,
  FILE_INPUT: `file-input-${baseTestID}`,
  SELECT_FILE_BUTTON: `select-file-button-${baseTestID}`,
  REMOVE_SELECTED_FILE_BUTTON: `remove-selected-file-button-${baseTestID}`
}

/**
 * Represent a file entry of specific type and a selected file
 * The data-filetype attribute is used to identify the file type
 * @constructor
 */

export const FileEntry = (props: FileEntryProps) => {
  const [selectedFile, setSelectedFile] = useState<File | null>(null)

  const uniqueId: string = generateUniqueId(); // unique id for the input element to ensure the onClick will find the correct element across all entries in the dom
  const fileTypeName = mapFileTypeToName(props.fileType);
  const fileChangedHandler = (e: ChangeEvent<HTMLInputElement>) => {
    const newFile: File = e.target?.files![0];
    if (newFile) {
      updateSelectedFile(newFile);
    }
  }

  const fileRemovedHandler = () => {
    updateSelectedFile(null);
  }

  const updateSelectedFile = (file: File | null) => {
    setSelectedFile(file); // update internal state
    // notify the parent component if it has provided handler
    if (props.notifySelectedFileChange) {
      props.notifySelectedFileChange(props.fileType, file);
    }
  }

  const debounceFileRemoveHandler = debounce(fileRemovedHandler,DEBOUNCE_INTERVAL)

  return <div data-filetype={props.fileType} data-testid={DATA_TEST_ID.FILE_ENTRY}>
    {
      selectedFile ?
        <Fab {...commonFabProps} {...selectedFileFabProps} color='secondary'
             aria-label={`Remove ${fileTypeName} csv file`}
             data-testid={DATA_TEST_ID.REMOVE_SELECTED_FILE_BUTTON}
             onClick={debounceFileRemoveHandler}>
          <RemoveIcon/>
          <span style={{
            display: 'inline-block',
            maxWidth: 'calc(100% - 20px)',
            overflow: 'hidden',
            textOverflow: 'ellipsis',
          }}>{fileTypeName}: {selectedFile.name}</span>
        </Fab>
        : <div>
          <input id={uniqueId} type='file' style={{display: 'none'}} datatype='.csv'
                 data-testid={DATA_TEST_ID.FILE_INPUT}
                 onChange={fileChangedHandler} data-filetype={props.fileType}/>
          <Fab {...commonFabProps} color='primary'
               aria-label={`Add ${fileTypeName} csv file`}
               data-testid={DATA_TEST_ID.SELECT_FILE_BUTTON}
               onClick={() => document.getElementById(uniqueId)!.click()}>
            <AddIcon/>{fileTypeName}
          </Fab>
        </div>
    }
  </div>;
}
export default FileEntry;

const commonFabProps = {
  size: "small" as FabProps['size'],
  component: "span",
  variant: "extended" as FabProps['variant'],
  sx: {textTransform: 'none'},
  // Add any other shared styles here
};
const selectedFileFabProps = {
  sx: {
    textTransform: 'none',
    justifyContent: 'flex-start',
    whiteSpace: 'nowrap',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    width: '250px'
  }
}