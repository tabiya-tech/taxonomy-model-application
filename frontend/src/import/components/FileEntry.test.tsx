// mute the console
import 'src/_test_utilities/consoleMock';

import {fireEvent, render, screen, waitFor} from "src/_test_utilities/test-utils";
import {DATA_TEST_ID, FileEntry} from "./FileEntry";
import ImportAPISpecs from "api-specifications/import";
import {mapFileTypeToName} from "./mapFileTypeToName";
import {clickDebouncedButton} from "src/_test_utilities/userEventFakeTimer";

describe("FileEntry render tests", () => {

  beforeEach(() => {
    (console.error as jest.Mock).mockClear();
    (console.warn as jest.Mock).mockClear();
  });

  it.each(
    [...Object.values(ImportAPISpecs.Constants.ImportFileTypes).map((fileType) => [fileType as ImportAPISpecs.Constants.ImportFileTypes, mapFileTypeToName(fileType)] as [ImportAPISpecs.Constants.ImportFileTypes, string])]
  )
  ("should render default state %s -> %s", (fileType: ImportAPISpecs.Constants.ImportFileTypes, expectedFileTypeName: string) => {
    // WHEN fileEntry is rendered with some fileType
    render(<FileEntry fileType={fileType}/>)

    // THEN expect no errors or warning to have occurred
    expect(console.error).not.toHaveBeenCalled();
    expect(console.warn).not.toHaveBeenCalled();
    // AND the component to be rendered
    const fileEntry = screen.getByTestId(DATA_TEST_ID.FILE_ENTRY);
    expect(fileEntry).toBeInTheDocument();
    // AND to have the correct file type
    expect(fileEntry).toHaveAttribute("data-filetype", fileType);

    // AND expect file input's to have no files
    const fileInput: HTMLInputElement = screen.getByTestId(DATA_TEST_ID.FILE_INPUT);
    expect(fileInput.files).toHaveLength(0);
    // AND expect file input to have the file type
    expect(fileInput).toHaveAttribute("data-filetype", fileType);

    // AND expect file trigger fab to be in the document
    const selectFileButton = screen.getByTestId(DATA_TEST_ID.SELECT_FILE_BUTTON)
    expect(selectFileButton).toBeInTheDocument();

    // AND expect file remover fab to not be in the document
    const fileRemoverButton = screen.queryByTestId(DATA_TEST_ID.REMOVE_SELECTED_FILE_BUTTON)
    expect(fileRemoverButton).not.toBeInTheDocument();

    // AND expect the SELECT_FILE_BUTTON to have the correct file type name
    expect(selectFileButton.textContent).toBe(expectedFileTypeName);
  })

  it("multiple components should have a unique id", () => {
    // GIVEN some filetype
    const givenFileType = ImportAPISpecs.Constants.ImportFileTypes.ESCO_OCCUPATION;

    // WHEN fileEntry is rendered multiples
    render(<FileEntry fileType={givenFileType}/>)
    render(<FileEntry fileType={givenFileType}/>)

    // THEN expect no errors or warning to have occurred
    expect(console.error).not.toHaveBeenCalled();
    expect(console.warn).not.toHaveBeenCalled();
    // AND to find two inputEntries
    let inputEntries = screen.getAllByTestId(DATA_TEST_ID.FILE_INPUT)
    expect(inputEntries.length).toBe(2);

    // AND their ids are different
    expect(inputEntries[0].id).not.toBe(inputEntries[1].id);
  })
})

describe("FileEntry action tests", () => {

  it("should selected file", async () => {
    // GIVEN some fileType
    const givenFileType = ImportAPISpecs.Constants.ImportFileTypes.ESCO_OCCUPATION;
    // AND some file
    const givenAFile = new File([], "foo.csv", {type: "text/csv"})

    // WHEN fileEntry is rendered
    render(<FileEntry fileType={givenFileType}/>)
    // AND fileInput value is changed
    const fileInput: HTMLInputElement = screen.getByTestId(DATA_TEST_ID.FILE_INPUT);
    fireEvent.change(fileInput, {target: {files: [givenAFile]}})

    // THEN expect file input to have 1 files
    expect(fileInput.files).toHaveLength(1);

    // AND expect file trigger fab to not be in the document
    const selectFileButton = screen.queryByTestId(DATA_TEST_ID.SELECT_FILE_BUTTON)
    expect(selectFileButton).not.toBeInTheDocument();

    // AND expect file remover fab to be in the document
    const fileRemoverFab = screen.queryByTestId(DATA_TEST_ID.REMOVE_SELECTED_FILE_BUTTON)
    expect(fileRemoverFab).toBeInTheDocument();
  })

  it("should remove selected file", async () => {
    // GIVEN some fileType
    const givenFileType = ImportAPISpecs.Constants.ImportFileTypes.ESCO_OCCUPATION;
    // AND some file
    const givenAFile = new File([], "foo.csv", {type: "text/csv"})
    // AND fileEntry is rendered
    render(<FileEntry fileType={givenFileType}/>)
    // AND fileInput value has changed
    let fileInput: HTMLInputElement = screen.getByTestId(DATA_TEST_ID.FILE_INPUT);
    fireEvent.change(fileInput, {target: {files: [givenAFile]}});
    expect(fileInput.files).toHaveLength(1);

    // WHEN fileRemoverFab is clicked
    const fileRemoverFab = screen.getByTestId(DATA_TEST_ID.REMOVE_SELECTED_FILE_BUTTON)
    await clickDebouncedButton(fileRemoverFab);

    // THEN expect file input to have an empty files
    // wait for the component's state to update
    await waitFor(() => {
      fileInput = screen.getByTestId(DATA_TEST_ID.FILE_INPUT);
    }, {});
    expect(fileInput.files).toHaveLength(0)
    // AND expect file trigger fab to be in the document
    const selectFileButton = screen.getByTestId(DATA_TEST_ID.SELECT_FILE_BUTTON)
    expect(selectFileButton).toBeInTheDocument()
  })

  it("should correctly notify the notifySelectedFileChange handler when file is selected", async () => {
    // GIVEN some fileType
    const givenFileType = ImportAPISpecs.Constants.ImportFileTypes.ESCO_OCCUPATION;
    // AND a notification handler
    const givenMockNotification = jest.fn();
    // AND some file
    const givenFile = new File([], "foo.csv", {type: "text/csv"})

    // WHEN fileEntry is rendered
    render(<FileEntry fileType={givenFileType} notifySelectedFileChange={givenMockNotification}/>);
    // AND a file is chosen
    const fileInput = screen.getByTestId(DATA_TEST_ID.FILE_INPUT);
    fireEvent.change(fileInput, {target: {files: [givenFile]}});

    // THEN expect notification to have been called with the given filetype and file
    expect(givenMockNotification).toHaveBeenCalledWith(givenFileType, givenFile);
  })

  it("should correctly notify the notifySelectedFileChange handler when file is removed", async () => {
    // GIVEN some fileType
    const givenFileType = ImportAPISpecs.Constants.ImportFileTypes.ESCO_OCCUPATION;
    // AND someNotificationHandler
    const givenMockNotification = jest.fn();
    // AND some file
    const givenAFile = new File([], "foo.csv", {type: "text/csv"})
    // AND fileEntry is rendered
    render(<FileEntry fileType={givenFileType} notifySelectedFileChange={givenMockNotification}/>)
    // AND the given file has been selected
    let fileInput: HTMLInputElement = screen.getByTestId(DATA_TEST_ID.FILE_INPUT);
    fireEvent.change(fileInput, {target: {files: [givenAFile]}});
    expect(fileInput.files).toHaveLength(1);

    // WHEN the remove selected file button is clicked
    const fileRemoverFab = screen.getByTestId(DATA_TEST_ID.REMOVE_SELECTED_FILE_BUTTON)
    // wait for the component's state to update
    await clickDebouncedButton(fileRemoverFab);

    // THEN expect notificationHandler to have been called with givenFileType and null,
    expect(givenMockNotification).toBeCalledWith(givenFileType, null);
  });

  it("should handle file changes even if notifySelectedFileChange handler is not set", async () => {
    // GIVEN some fileType
    const givenFileType = ImportAPISpecs.Constants.ImportFileTypes.ESCO_OCCUPATION;
    // AND some file
    const givenAFile = new File([], "foo.csv", {type: "text/csv"})

    // When  fileEntry is rendered without a notification handler
    render(<FileEntry fileType={givenFileType}/>)
    // AND fileInput value has changed
    let fileInput: HTMLInputElement = screen.getByTestId(DATA_TEST_ID.FILE_INPUT);
    fireEvent.change(fileInput, {target: {files: [givenAFile]}});

    // THEN expect file input to have 1 files
    expect(fileInput.files).toHaveLength(1);

    // AND expect the fileRemoverButton to be in the document
    const fileRemoverButton = screen.getByTestId(DATA_TEST_ID.REMOVE_SELECTED_FILE_BUTTON)
    expect(fileRemoverButton).toBeInTheDocument();
  });
});