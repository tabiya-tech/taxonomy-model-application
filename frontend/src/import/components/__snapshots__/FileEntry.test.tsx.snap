// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileEntry render tests should render default state ESCO_SKILL_GROUPS -> Skill Groups 1`] = `
<div
  data-filetype="ESCO_SKILL_GROUPS"
  data-testid="file-entry-d2bc4d5d-7760-450d-bac6-a8857affeb89"
>
  <div>
    <input
      accept=".csv"
      data-filetype="ESCO_SKILL_GROUPS"
      data-testid="file-input-d2bc4d5d-7760-450d-bac6-a8857affeb89"
      id="ESCO_SKILL_GROUPS"
      style="display: none;"
      type="file"
    />
    <div
      aria-label="Add Skill Groups csv file"
      class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-clickable MuiChip-clickableColorPrimary MuiChip-filledPrimary css-d8rrhm-MuiButtonBase-root-MuiChip-root"
      data-testid="select-file-button-d2bc4d5d-7760-450d-bac6-a8857affeb89"
      role="button"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorPrimary css-etbowg-MuiSvgIcon-root"
        data-testid="AddCircleOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
        />
      </svg>
      <span
        class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
      >
        Skill Groups
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
  </div>
</div>
`;

exports[`FileEntry render tests should render default state ESCO_SKILL_HIERARCHY -> Skill Hierarchy 1`] = `
<div
  data-filetype="ESCO_SKILL_HIERARCHY"
  data-testid="file-entry-d2bc4d5d-7760-450d-bac6-a8857affeb89"
>
  <div>
    <input
      accept=".csv"
      data-filetype="ESCO_SKILL_HIERARCHY"
      data-testid="file-input-d2bc4d5d-7760-450d-bac6-a8857affeb89"
      id="ESCO_SKILL_HIERARCHY"
      style="display: none;"
      type="file"
    />
    <div
      aria-label="Add Skill Hierarchy csv file"
      class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-clickable MuiChip-clickableColorPrimary MuiChip-filledPrimary css-d8rrhm-MuiButtonBase-root-MuiChip-root"
      data-testid="select-file-button-d2bc4d5d-7760-450d-bac6-a8857affeb89"
      role="button"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorPrimary css-etbowg-MuiSvgIcon-root"
        data-testid="AddCircleOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
        />
      </svg>
      <span
        class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
      >
        Skill Hierarchy
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
  </div>
</div>
`;

exports[`FileEntry render tests should render default state ESCO_SKILL_SKILL_RELATIONS -> Skill-To-Skill Relations 1`] = `
<div
  data-filetype="ESCO_SKILL_SKILL_RELATIONS"
  data-testid="file-entry-d2bc4d5d-7760-450d-bac6-a8857affeb89"
>
  <div>
    <input
      accept=".csv"
      data-filetype="ESCO_SKILL_SKILL_RELATIONS"
      data-testid="file-input-d2bc4d5d-7760-450d-bac6-a8857affeb89"
      id="ESCO_SKILL_SKILL_RELATIONS"
      style="display: none;"
      type="file"
    />
    <div
      aria-label="Add Skill-To-Skill Relations csv file"
      class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-clickable MuiChip-clickableColorPrimary MuiChip-filledPrimary css-d8rrhm-MuiButtonBase-root-MuiChip-root"
      data-testid="select-file-button-d2bc4d5d-7760-450d-bac6-a8857affeb89"
      role="button"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorPrimary css-etbowg-MuiSvgIcon-root"
        data-testid="AddCircleOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
        />
      </svg>
      <span
        class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
      >
        Skill-To-Skill Relations
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
  </div>
</div>
`;

exports[`FileEntry render tests should render default state ESCO_SKILLS -> Skills 1`] = `
<div
  data-filetype="ESCO_SKILLS"
  data-testid="file-entry-d2bc4d5d-7760-450d-bac6-a8857affeb89"
>
  <div>
    <input
      accept=".csv"
      data-filetype="ESCO_SKILLS"
      data-testid="file-input-d2bc4d5d-7760-450d-bac6-a8857affeb89"
      id="ESCO_SKILLS"
      style="display: none;"
      type="file"
    />
    <div
      aria-label="Add Skills csv file"
      class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-clickable MuiChip-clickableColorPrimary MuiChip-filledPrimary css-d8rrhm-MuiButtonBase-root-MuiChip-root"
      data-testid="select-file-button-d2bc4d5d-7760-450d-bac6-a8857affeb89"
      role="button"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorPrimary css-etbowg-MuiSvgIcon-root"
        data-testid="AddCircleOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
        />
      </svg>
      <span
        class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
      >
        Skills
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
  </div>
</div>
`;

exports[`FileEntry render tests should render default state ISCO_GROUPS -> ISCO Groups 1`] = `
<div
  data-filetype="ISCO_GROUPS"
  data-testid="file-entry-d2bc4d5d-7760-450d-bac6-a8857affeb89"
>
  <div>
    <input
      accept=".csv"
      data-filetype="ISCO_GROUPS"
      data-testid="file-input-d2bc4d5d-7760-450d-bac6-a8857affeb89"
      id="ISCO_GROUPS"
      style="display: none;"
      type="file"
    />
    <div
      aria-label="Add ISCO Groups csv file"
      class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-clickable MuiChip-clickableColorPrimary MuiChip-filledPrimary css-d8rrhm-MuiButtonBase-root-MuiChip-root"
      data-testid="select-file-button-d2bc4d5d-7760-450d-bac6-a8857affeb89"
      role="button"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorPrimary css-etbowg-MuiSvgIcon-root"
        data-testid="AddCircleOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
        />
      </svg>
      <span
        class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
      >
        ISCO Groups
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
  </div>
</div>
`;

exports[`FileEntry render tests should render default state OCCUPATION_HIERARCHY -> Occupation Hierarchy 1`] = `
<div
  data-filetype="OCCUPATION_HIERARCHY"
  data-testid="file-entry-d2bc4d5d-7760-450d-bac6-a8857affeb89"
>
  <div>
    <input
      accept=".csv"
      data-filetype="OCCUPATION_HIERARCHY"
      data-testid="file-input-d2bc4d5d-7760-450d-bac6-a8857affeb89"
      id="OCCUPATION_HIERARCHY"
      style="display: none;"
      type="file"
    />
    <div
      aria-label="Add Occupation Hierarchy csv file"
      class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-clickable MuiChip-clickableColorPrimary MuiChip-filledPrimary css-d8rrhm-MuiButtonBase-root-MuiChip-root"
      data-testid="select-file-button-d2bc4d5d-7760-450d-bac6-a8857affeb89"
      role="button"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorPrimary css-etbowg-MuiSvgIcon-root"
        data-testid="AddCircleOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
        />
      </svg>
      <span
        class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
      >
        Occupation Hierarchy
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
  </div>
</div>
`;

exports[`FileEntry render tests should render default state OCCUPATION_SKILL_RELATIONS -> Occupation-To-Skill Relations 1`] = `
<div
  data-filetype="OCCUPATION_SKILL_RELATIONS"
  data-testid="file-entry-d2bc4d5d-7760-450d-bac6-a8857affeb89"
>
  <div>
    <input
      accept=".csv"
      data-filetype="OCCUPATION_SKILL_RELATIONS"
      data-testid="file-input-d2bc4d5d-7760-450d-bac6-a8857affeb89"
      id="OCCUPATION_SKILL_RELATIONS"
      style="display: none;"
      type="file"
    />
    <div
      aria-label="Add Occupation-To-Skill Relations csv file"
      class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-clickable MuiChip-clickableColorPrimary MuiChip-filledPrimary css-d8rrhm-MuiButtonBase-root-MuiChip-root"
      data-testid="select-file-button-d2bc4d5d-7760-450d-bac6-a8857affeb89"
      role="button"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorPrimary css-etbowg-MuiSvgIcon-root"
        data-testid="AddCircleOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
        />
      </svg>
      <span
        class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
      >
        Occupation-To-Skill Relations
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
  </div>
</div>
`;

exports[`FileEntry render tests should render default state OCCUPATIONS -> Occupations 1`] = `
<div
  data-filetype="OCCUPATIONS"
  data-testid="file-entry-d2bc4d5d-7760-450d-bac6-a8857affeb89"
>
  <div>
    <input
      accept=".csv"
      data-filetype="OCCUPATIONS"
      data-testid="file-input-d2bc4d5d-7760-450d-bac6-a8857affeb89"
      id="OCCUPATIONS"
      style="display: none;"
      type="file"
    />
    <div
      aria-label="Add Occupations csv file"
      class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-clickable MuiChip-clickableColorPrimary MuiChip-filledPrimary css-d8rrhm-MuiButtonBase-root-MuiChip-root"
      data-testid="select-file-button-d2bc4d5d-7760-450d-bac6-a8857affeb89"
      role="button"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorPrimary css-etbowg-MuiSvgIcon-root"
        data-testid="AddCircleOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
        />
      </svg>
      <span
        class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
      >
        Occupations
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
  </div>
</div>
`;
